docs/iris/example_code/General/coriolis_plot.py

```diff
@@ -0,0 +1,78 @@
+"""
+Deriving the Coriolis frequency over the globe
+==============================================
+
+This code computes Coriolis frequency and stores them in a cube with
+associated metadata.
+
+It plots the Coriolis frequency on an orthographic projection.
+
+"""
+
+import cartopy.crs as ccrs
+import iris
+from iris.coord_systems import GeogCS
+import iris.plot as iplt
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def main():
+    # Start with arrays for latitudes and longitudes, with a given resolution.
+    coordinate_points = 200
+    longitudes = np.linspace(-180.0, 180.0, coordinate_points)
+    latitudes = np.linspace(-90.0, 90.0, coordinate_points)
+    lon2d, lat2d = np.meshgrid(longitudes, latitudes)
+
+    # Omega is the Earth's rotation rate, expressed in radians per second
+    omega = 7.29e-5
+
+    # The data for our cube is the Coriolis frequency,
+    # `f = 2 * omega * sin(phi)`, which is computed for each grid point over
+    # the globe from the 2-dimensional latitude array.
+    data = 2. * omega * np.sin(np.deg2rad(lat2d))
+
+    # We now need to define a coordinate system for the plot.
+    # Here we'll use GeogCS; 6371229 is the radius of the Earth in metres.
+    cs = GeogCS(6371229)
+
+    # The Iris coords module turns the latitude list into a coordinate array.
+    # Coords then applies an appropriate standard name and unit to it.
+    lat_coord = iris.coords.DimCoord(latitudes,
+                                     standard_name='latitude',
+                                     units='degrees',
+                                     coord_system=cs)
+
+    # The above process is repeated for the longitude coordinates.
+    lon_coord = iris.coords.DimCoord(longitudes,
+                                     standard_name='longitude',
+                                     units='degrees',
+                                     coord_system=cs)
+
+    # Now we input our data array into the cube.
+    new_cube = iris.cube.Cube(data,
+                              standard_name='coriolis_parameter',
+                              units='s-1',
+                              dim_coords_and_dims=[(lat_coord, 0),
+                                                   (lon_coord, 1)])
+
+    # This section checks to ensure that the coordinates are bounded.
+    # If they have no bounds, the code applies some via guess_bounds()
+    for coord in new_cube.coords():
```

Comment:
This is the only problematic part left. I'm not totally opposed to calling `guess_bounds()` as long as it is explained properly. You should also avoid the loop and the isinstance check. Again, you are creating from scratch so you don't need to inspect the coordinates, you made them so you know if they have bounds or not! Perhaps this could be moved to before the cube is created and be restructured as in:

```python
# Add bounds to our latitude and longitude coordinates. We want simple
# contiguous bounds for our regularly spaced coordinate points so we just
# use the guess_bounds() method of the coordinate which does exactly this.
# If our coordinates were more complex we could derive and set the bounds
# manually.
lat_coord.guess_bounds()
lon_coord.guess_bounds()
```

This leaves me with only 1 minor concern, you latitude coordinate includes the poles and therefore the generated bounds may lie outside the valid [-90,90] range. I think @rhattersley fixed this in #1978 though... so maybe this is fine now.